docs/content/docs/deployment/elastic_scaling.md

```diff
@@ -191,6 +191,8 @@ Adaptive Batch Scheduler will only decide parallelism for operators whose parall
 
 - **Batch jobs only**: Adaptive Batch Scheduler only supports batch jobs. Exception will be thrown if a streaming job is submitted.
 - **ALL-EXCHANGES-BLOCKING jobs only**: At the moment, Adaptive Batch Scheduler only supports jobs whose [shuffle mode]({{< ref "docs/deployment/config" >}}#execution-batch-shuffle-mode) is `ALL-EXCHANGES-BLOCKING`.
+- **The decided parallelism can only be a power of 2**: In order to make the subpartitoins evenly consumed by downstream tasks, user should configure the [`jobmanager.adaptive-batch-scheduler.max-parallelism`]({{< ref "docs/deployment/config" >}}#jobmanager-adaptive-batch-scheduler-max-parallelism) to be a power of 2 (2^N), and the decided parallelism will also be a power of 2 (2^M and M < N).
+- **No support for serveral file APIs**: No support for `StreamExecutionEnvironment#readFile` `StreamExecutionEnvironment#readTextFile` `StreamExecutionEnvironment#createInput(FileInputFormat, ...)` and all data sources using these APIs. When using these APIs, there will be a separate monitoring task (called a `Custom File Source`) as a predecessor to the actual data sources, which Adaptive Batch Scheduler cannot handle.
```

Comment:
serveral -> several

Reply:
> No support for serveral file APIs

maybe "FileInputFormat sources are not supported", and later in the description state that "FileInputFormat sources include `StreamExecutionEnvironment#readFile(...)`, `StreamExecutionEnvironment#readTextFile(...)` and`StreamExecutionEnvironment#createInput(FileInputFormat, ...)`"

Reply:
> When using these APIs, there will be a separate monitoring task (called a `Custom File Source`) as a predecessor to the actual data sources, which Adaptive Batch Scheduler cannot handle.

Maybe "FileInputFormat sources implementation requires the parallelism to be decided in ahead, which is against the rule of Adaptive Batch Scheduler."?

Reply:
I would also point users to the new sources (`StreamExecutionEnvironment#fromSource(...)`) or `flink-connector-files`(with a link) for file reading when using Adaptive Batch Scheduler.